lib/ask/runtime/questionnaire_simulator.ex

```diff
@@ -124,8 +138,9 @@ defmodule Ask.Simulation.SubmittedStep do
     responses = Reply.stores(reply)
                 |> Enum.map(fn {step_name, value} ->
                   referred_step = questionnaire |> Questionnaire.all_steps |> Enum.filter(fn step -> step["store"] == step_name end)
-                  [id] = referred_step |> Enum.map(fn step -> step["id"] end)
-                  [title] = referred_step |> Enum.map(fn step -> step["title"] end)
+                  # hd function is used since there is a restriction that two steps cannot have the same store variable name
+                  id = referred_step |> Enum.map(fn step -> step["id"] end) |> hd
+                  title = referred_step |> Enum.map(fn step -> step["title"] end) |> hd
```

Comment:
If I'm not missing something here, if we replace `filter` with `find` neither `map` nor `hd` are needed
```suggestion
                  referred_step = questionnaire |> Questionnaire.all_steps |> Enum.find(fn step -> step["store"] == step_name end)
                  %{"title" => title, "id" => id} = referred_step
```

Reply:
You are so right! thanks for spotting it! 🙌